src/Databases/DatabaseAtomic.cpp

```diff
@@ -571,12 +571,15 @@ void DatabaseAtomic::renameDictionaryInMemoryUnlocked(const StorageID & old_name
     const auto & dict = dynamic_cast<const IDictionary &>(*result.object);
     dict.updateDictionaryName(new_name);
 }
-void DatabaseAtomic::waitDetachedTableNotInUse(const UUID & uuid)
+void DatabaseAtomic::waitDetachedTableNotInUseOrThrow(const UUID & uuid, bool throw_if_in_use)
 {
+    if (throw_if_in_use)
```

Comment:
Looks like this is literally two separate functions in one, chosen by a bool flag. Why not make them separate?